components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -263,7 +263,13 @@ public Response authorize(@Context HttpServletRequest request, @Context HttpServ
         try {
             // Start tenant domain flow if the tenant configuration is not enabled.
             if (!IdentityTenantUtil.isTenantedSessionsEnabled()) {
-                String tenantDomain = EndpointUtil.getSPTenantDomainFromClientId(oAuthMessage.getClientId());
+                String tenantDomain = null;
+                if (StringUtils.isNotEmpty(oAuthMessage.getClientId())) {
+                    tenantDomain = EndpointUtil.getSPTenantDomainFromClientId(oAuthMessage.getClientId());
+                } else if (oAuthMessage.getSessionDataCacheEntry() != null) {
+                    OAuth2Parameters oauth2Params = getOauth2Params(oAuthMessage);
+                    tenantDomain = oauth2Params.getTenantDomain();
```

Comment:
can there be any instance where oAuthMessage.getClientId()  is empty and OAuthMessage.getSessionDataCacheEntry() is null?

Tenant domain will be null in that case

Reply:
as the client id is a mandatory parameter for the authorize call, the first requests will go in that block. 

When the authentication is completed, the flow will go in the else if block. In that case, the SessionDataCacheEntry should be available, else remaining flows will also fail.